src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx

```diff
@@ -50,33 +54,29 @@ class JupyterLiteTransformation extends React.Component<
             // eslint-disable-next-line react/no-did-update-set-state
             this.setState({ materials });
         }
+        this.messageHandler.sendData(this.returnSelectedMaterials());
     }
 
-    componentWillUnmount() {
-        window.removeEventListener("message", this.handleReceiveMessage, false);
-    }
+    returnSelectedMaterials = () => {
+        const { selectedMaterials } = this.state;
+        return selectedMaterials.map((material) => material.toJSON());
+    };
 
-    handleReceiveMessage = (event: MessageEvent) => {
-        // Check if the message is from the expected source
-        // TODO: check for partial URL match, e.g. with "/" at the end
-        if (event.origin !== ORIGIN_URL) {
-            return;
-        }
-        if (event.data.type === "from-iframe-to-host") {
-            try {
-                // TODO: add check for Material config type
-                const configs = event.data.data.materials;
-                if (Array.isArray(configs)) {
-                    this.setState({
-                        newMaterials: configs.map((config) => new Made.Material(config)),
-                    });
-                }
-            } catch (err) {
-                console.log(err);
-            }
-            if (event.data.requestData === true && event.data.variableName === "materials_in") {
-                this.sendMaterialsToIFrame();
+    setMaterials = (data: any) => {
+        // Since any arbitrary data can be sent from the JupyterLite notebook, the validation might fail
+        try {
+            const configs = data.materials as MaterialSchema[];
+            if (Array.isArray(configs)) {
+                this.setState({
+                    newMaterials: configs.map((config) => {
+                        const m = new Made.Material(config);
+                        m.validate();
```

Comment:
Let's show an error message to the user?? Using notistack.

The logic should be:

1. Run validation for all configs
2. Throw exception and show user a message explaining which item is incorrect and why
3. Filter out the configs that don't pass validation
4. Proceed with the rest of the configs to set the state